File: 1500-count-largest-group/1500-count-largest-group.cpp
class Solution {
public:
    int findDigitSum(int n){
        int sm = 0;
        while(n){
            sm += n%10;
            n = n/10;
        }
        return sm;
    }
    int countLargestGroup(int n) {
        int maxSize = 0;
        int count = 0;
        unordered_map<int,int> mp;
        for(int num = 1; num <= n; num++){
            int sm = findDigitSum(num);
            mp[sm]++;
            if(mp[sm] == maxSize){
                count++;
            }else if(mp[sm] > maxSize){
                maxSize = mp[sm];
                count = 1;
            }
        }
        return count;
    }
};

File: 1500-count-largest-group/1500-count-largest-group.java
class Solution {
    int findDigitSum(int num){
        int sum = 0;
        while(num > 0){
            sum += num%10;
            num /= 10;
        }
        return sum;
    }
    public int countLargestGroup(int n) {
        int[] mp = new int[40];
        int max = 0;
        for(int num = 1; num <= n; num++){
            int sum = findDigitSum(num);
            mp[sum]++;
            if(mp[sum] > max){
                max = mp[sum];
            }
        }
        int count = 0;
        for(int i = 0; i<40; i++){
            if(mp[i] == max){
                count++;
            }
        }
        return count;
    }
}